syntax = "proto3";

package rti;

message RequestModifyOrder
	{                                
        enum PriceType {
    PRICETYPE_UNSPECIFIED = 0;
	                LIMIT              =  1;
			MARKET             =  2;
			STOP_LIMIT         =  3;
			STOP_MARKET        =  4;
			MARKET_IF_TOUCHED  =  5;
			LIMIT_IF_TOUCHED   =  6;
                       }

	enum PriceField {
    PRICEFIELD_UNSPECIFIED = 0;
	                 BID_PRICE    =  1;
			 OFFER_PRICE  =  2;
			 TRADE_PRICE  =  3;
			 LEAN_PRICE   =  4;
	                }

	enum Condition {
    CONDITION_UNSPECIFIED = 0;
	     	        EQUAL_TO               =  1;
			NOT_EQUAL_TO           =  2;
			GREATER_THAN           =  3;
			GREATER_THAN_EQUAL_TO  =  4;
			LESSER_THAN            =  5;
			LESSER_THAN_EQUAL_TO   =  6;
	               }

        enum OrderPlacement {
    ORDERPLACEMENT_UNSPECIFIED = 0;
	                     MANUAL  = 1;
			     AUTO    = 2;
	                    }


	int32             template_id            = 154467;
	repeated string            user_msg               = 132760;
	string            window_name            = 154629;

	string            fcm_id                 = 154013;
	string            ib_id                  = 154014;
	string            account_id             = 154008;
	string            basket_id              = 110300;

	string            symbol                 = 110100;
	string            exchange               = 110101;

	 int32            quantity               = 112004;
	 double           price                  = 110306;
	 double           trigger_price          = 149247;

	 PriceType        price_type             = 112008;
	 OrderPlacement   manual_or_auto         = 154710;

	 string           if_touched_symbol      = 154451;
	 string           if_touched_exchange    = 154452;
	 Condition        if_touched_condition   = 154453;
	 PriceField       if_touched_price_field = 154454;
	 double           if_touched_price       = 153632;
	}
